flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/ScanContext.java

```diff
@@ -114,77 +128,104 @@ private ScanContext(boolean caseSensitive, Long snapshotId, Long startSnapshotId
     this.includeColumnStats = includeColumnStats;
     this.exposeLocality = exposeLocality;
     this.planParallelism = planParallelism;
+
+    validate();
+  }
+
+  private void validate() {
+    if (isStreaming) {
```

Comment:
Will it be allowed to set any `startingStrategy` value if it's in batch mode ?

Reply:
we can add the validation that `startingStrategy` is null for batch mode.

will also need to change the default value to null for the config option.
```
  private static final ConfigOption<StreamingStartingStrategy> STARTING_STRATEGY =
      ConfigOptions.key("starting-strategy").enumType(StreamingStartingStrategy.class)
          .defaultValue(StreamingStartingStrategy.LATEST_SNAPSHOT);
```